### ISSUE TITLE
Tooltip overflow

### ISSUE THREAD

**Issue by hsestupin** [convergencelabs/monaco-collab-ext#8]
When remote cursor is added at the last line of the editor (see red cursor at the screen) the tooltip is hidden:
![Screenshot from 2020-07-28 18-51-16](https://user-images.githubusercontent.com/1468392/88696449-926dd500-d103-11ea-9353-9562fda77a97.png)

Obviously it happens only when there is not enough space at the bottom editor. But I guess the behaviour in this particular case should be the same like with a completion menu. Tooltip shouldn't be cut by the editor borders. 

There is a flag `IContentWidget.allowEditorOverflow` which could help to resolve the problem, however I'm not sure.

**Comment by mmacfadden** [comment]
Sorry for the long delay. In our other editors we make the tooltip change to place above the cursor when it is near the bottom.  I will take a look.

**Comment by igRo** [comment]
In some cases placing the tooltip above the caret won't fix the issue (you can see that remote cursor is also trimmed on the left):
<img width="152" alt="Screenshot 2021-04-08 at 17 56 55" src="https://user-images.githubusercontent.com/376776/114058769-0c11b180-9894-11eb-86ba-af3407323836.png">

Completion works fine under the same conditions:
<img width="486" alt="Screenshot 2021-04-08 at 17 57 55" src="https://user-images.githubusercontent.com/376776/114058842-1cc22780-9894-11eb-9218-02a9cfc40192.png">

**Comment by mmacfadden** [comment]
Yes. I think the intended behavior would be that the remote cursor moves up / down and left / right to avoid the border of the editor.
